### PR TITLE
[tests-only]Add test for ANTIVIRUS_INFECTED_FILE_HANDLING

### PR DESCRIPTION
**Pull request by S-Panta**
## Description
This PR adds API test for antivirus with various environment variable setting. The added scenarios are

- upload a file with virus with various infected file handling setting set to continue

## Related Issue

- Part of <https://github.com/owncloud/ocis/issues/6255>

## How Has This Been Tested?

- test environment:
- locally
- CI

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: tests/acceptance/features/apiAntivirus/antivirus.feature [hunk 1]
```diff
@@ -127,3 +127,35 @@ Feature: antivirus
       | old              |
       | new              |
       | spaces           |
+
+  @env-config
+  Scenario Outline: upload a file with virus with various infected file handling setting set to continue
```
Comment (SagarGi) [inline tests/acceptance/features/apiAntivirus/antivirus.feature:132]:
```suggestion
  Scenario Outline: upload a virus infected file along with various infected file handling setting set to continue
```
if this seems correct to you.

File: tests/acceptance/features/apiAntivirus/antivirus.feature [hunk 2]
```diff
@@ -127,3 +127,35 @@ Feature: antivirus
...
+      | Virus found in aFileWithVirus.txt. Upload not possible. Virus: Win.Test.EICAR_HDB-1 |
+    And as "Alice" file "/aFileWithVirus.txt" should exist
+    Examples:
+      | dav-path-version |
+      | old              |
+      | new              |
+      | spaces           |
+
+  @env-config
+  Scenario Outline: upload a file with virus with various infected file handling setting set to delete
```
Comment (SagarGi) [inline tests/acceptance/features/apiAntivirus/antivirus.feature:148]:
```suggestion
  Scenario Outline: upload a virus infected file along with various infected file handling setting set to delete
```

File: tests/acceptance/features/apiAntivirus/antivirus.feature (hunk 1 above)
Comment (grgprarup) [reply tests/acceptance/features/apiAntivirus/antivirus.feature:132]:
```suggestion
  Scenario Outline: upload a file with the virus with config ANTIVIRUS_INFECTED_FILE_HANDLING set to continue
```
IMO, the scenario is related to setting the config to continue.

File: tests/acceptance/features/apiAntivirus/antivirus.feature [hunk 3]
```diff
@@ -127,3 +127,35 @@ Feature: antivirus
...
+  @env-config
+  Scenario Outline: upload a file with virus with various infected file handling setting set to continue
+    Given the config "ANTIVIRUS_INFECTED_FILE_HANDLING" has been set to "continue"
+    And using <dav-path-version> DAV path
+    When user "Alice" uploads file "filesForUpload/filesWithVirus/eicar.com" to "/aFileWithVirus.txt" using the WebDAV API
+    Then the HTTP status code should be "201"
+    And user "Alice" should get a notification with subject "Virus found" and message:
+      | message                                                                             |
+      | Virus found in aFileWithVirus.txt. Upload not possible. Virus: Win.Test.EICAR_HDB-1 |
+    And as "Alice" file "/aFileWithVirus.txt" should exist
```
Comment (grgprarup) [inline tests/acceptance/features/apiAntivirus/antivirus.feature:140]: The notification message says `Upload not possible`, but the file is uploaded. What might the expected message be?

File: tests/acceptance/features/apiAntivirus/antivirus.feature (hunk 2 above)
Comment (grgprarup) [reply tests/acceptance/features/apiAntivirus/antivirus.feature:148]:
```suggestion
  Scenario Outline: upload a file with the virus with config ANTIVIRUS_INFECTED_FILE_HANDLING set to delete
```

File: tests/acceptance/features/apiAntivirus/antivirus.feature (hunk 3 above)
Comment (SwikritiT) [reply tests/acceptance/features/apiAntivirus/antivirus.feature:140]: there's a bug report for this https://github.com/owncloud/ocis/issues/6494, need someone from owncloud to verify

Comment (SwikritiT) [reply tests/acceptance/features/apiAntivirus/antivirus.feature:140]: @ScharfViktor we have a misleading notification when, `ANTIVIRUS_INFECTED_FILE_HANDLING` is set to `continue` and a file with the virus is uploaded. There's an issue report https://github.com/owncloud/ocis/issues/6494 however what to do in this scenario? Do we not check the notification? or change the message to something like `Virus found in aFileWithVirus.txt. Upload successful. Virus: Win.Test.EICAR_HDB-1` and add the test in expected to fail or wait till the bug is fixed?

File: tests/acceptance/features/apiAntivirus/antivirus.feature [hunk 4]
```diff
@@ -212,3 +212,18 @@ Feature: antivirus
       | eicar.com     | virusFile1.txt |
       | eicar_com.zip | virusFile2.zip |
 
+  @env-config
+  Scenario Outline: upload a file with virus by setting antivirus infected file handling environment variable to continue
```
Comment (saw-jan) [inline tests/acceptance/features/apiAntivirus/antivirus.feature:216]: so much suggestions in a scenario description :sweat_smile: 
```suggestion
  Scenario Outline: upload a file with virus by setting antivirus infected_file_handling config to continue
```
Others LGTM

**Review by SagarGi — COMMENTED** [review]
LGTM :+1: please rebase and resolve conflict

**Comment by SwikritiT** [comment]
`ANTIVIRUS_INFECTED_FILE_HANDLING = "continue"` is just for debugging purposes see https://github.com/owncloud/ocis/issues/6494#issuecomment-1598524794

So we don't need to add api tests for this option.

Closing this here

**Comment by SwikritiT** [comment]
Let's add this test even though it's a debug option @S-Panta please remove the step where we check the notification and rebase and push

**Review by SwikritiT — APPROVED** [review]
LGTM

**Review by ScharfViktor — APPROVED** [review]

**Pull request merged**